### PR TITLE
Install pkg-config files

### PR DESCRIPTION
**Pull request by jakirkham**
This installs the pkg-config files for `ncurses`. This should make it easier to figure out how to link against it correctly.

xref: https://github.com/bioconda/bioconda-recipes/pull/1391#issuecomment-217665645
xref: https://github.com/conda-forge/ncurses-feedstock/issues/9

cc @croth1

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -76,6 +82,22 @@ test:
...
+        "ncurses.pc",
+        "panel.pc",
+        "tinfo.pc"
+    ] %}
+    {% for each_ncurses_pc_file in ncurses_pc_files %}
+    - test -f ${PREFIX}/lib/pkgconfig/{{ each_ncurses_pc_file }}
+    {% endfor %}
+
+    # Test ncurses library arguments.
+    - pkg-config ncurses --libs | grep "^-L${PREFIX}/lib -lncurses -ltinfo $"
```
Comment (jakirkham) [inline recipe/meta.yaml]: Verifies that `libncurses` and `libtinfo` will both be linked against when using `pkg-config`.

**Comment by jakirkham** [comment]
@croth1, do you want to take a look at this and see if it works for you? I added a test with `pkg-config`, which seems to show the right libraries are included for linking (namely `libncurses` and `libtinfo`).

**Comment by croth1** [comment]
Thanks for tackling this. Looks good to me!

**Comment by jakirkham** [comment]
So, one somewhat surprising thing to note here. `ncurses` does not have `pkg-config` files for widec. It simply doesn't include them. I'm really not sure why they don't. We could probably generate them somehow.

**Comment by jakirkham** [comment]
Looks like `ncurses` can generate them, but it wasn't because it thought that step was completed. Cleaning the build directory appears to fix that issue. This is something we probably should have been doing already anyways. Will update this PR with that change.

**Comment by jakirkham** [comment]
The cleaning step has been broken out into a separate PR ( https://github.com/conda-forge/ncurses-feedstock/pull/12 ), which this is rebased on top of.

**Comment by jakirkham** [comment]
Also, updated the tests here to work with `pkg-config` files in the widec case in addition to the non-widec case.

**Comment by jakirkham** [comment]
@croth1, could you please take another look? I made some changes to handle widec support. There are tests for both widec and non-widec cases. None of the changes I made should break anything. Still it would be good if you could confirm that everything still works.

@jjhelmus, could you also please take a look? This seems pretty simple and straightforward, but maybe I have missed something obvious that you will catch.

Unless there are reviewer comments that I need to address here, I don't plan on making any more changes as this seems to be working correctly in all the cases I have tried.

**Comment by bgruening** [comment]
:+1: @jakirkham thanks for working on it!

**Comment by jakirkham** [comment]
Absolutely, @bgruening.

**Comment by jakirkham** [comment]
Well, I'm not hearing any complaints over this. So, I think I'm just going to merge this and we can iterate from there. Thanks everyone.

**Comment by jakirkham** [comment]
This has been released and is available from `conda-forge` for Mac and Linux. 🎉

**Comment by jjhelmus** [comment]
LGTM, glad to see this got merged.  Sorry for being late to the party.

**Pull request merged**